### PR TITLE
test(e2e): add signup logic to e2e tests

### PR DESCRIPTION
**Pull request by tdelabro**

### CODE REVIEW

File: marketplace-tests/src/e2e_tests/contributors/mod.rs [hunk 1]
```diff
@@ -1,2 +1,7 @@
 pub mod contact_information;
+
 pub mod get;
+pub use get::get;
```
Comment (AnthonyBuisset) [inline marketplace-tests/src/e2e_tests/contributors/mod.rs:4]: I think we should choose between `pub mod` and `pub use` in cases like this:
option 1: `contributors::get()` and `contributors::get_by_account()`
option 2: `contributors::get::by_id()` and `contributors::get::by_account()`

File: marketplace-tests/Cargo.toml [hunk 2]
```diff
@@ -47,6 +48,7 @@ async-trait = "0.1.56"
 
 # Local dependencies
 marketplace-core = { path = "../marketplace-core" }
+marketplace-domain = { path = "../marketplace-domain" }
```
Comment (AnthonyBuisset) [inline marketplace-tests/Cargo.toml:51]: This looks like a smell to me, e2e tests should not know the domain. Also I do not see in your PR where it is used

File: marketplace-tests/src/e2e_tests/scenario/contribution_lifetime.rs [hunk 3]
```diff
@@ -39,32 +40,18 @@ async fn contribution_lifetime(
 	assert_eq!(contribution.status, "OPEN");
 
 	// Apply to the contribution
-	let contributor_id = String::from("0x0029");
-	contributions::apply(&contribution.id, &contributor_id).await;
+	let contributor_address = format!("{:#066x}", accounts[2].address());
+	contributions::apply(&contribution.id, &contributor_address).await;
```
Comment (AnthonyBuisset) [inline marketplace-tests/src/e2e_tests/scenario/contribution_lifetime.rs:44]: technically, the `apply` should come after the `signup`

File: marketplace-tests/src/e2e_tests/scenario/contribution_lifetime.rs [hunk 4]
```diff
@@ -39,32 +40,18 @@ async fn contribution_lifetime(
...
 
-	// TODO restore when user registration can be tested in local
-	// Get the contributor
-	/*
-	let contributor = contributors::get::get(contributor_account.clone()).await;
-	assert_eq!(contributor.id, contributor_account);
-	assert_eq!(contributor.account, contributor_account);
-	assert_eq!(contributor.github_identifier, "");
-	assert_eq!(contributor.github_username, "");
+	contributors::signup(accounts[2].clone()).await;
```
Comment (AnthonyBuisset) [inline marketplace-tests/src/e2e_tests/scenario/contribution_lifetime.rs:48]: can you name this variable (e.g. `contributor_account`) and use this variable above as well ?

File: scripts/docker/dev/httpmocks/github_access_token.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+when:
```
Comment (AnthonyBuisset) [inline scripts/docker/dev/httpmocks/github_access_token.yaml:1]: excellent, we should maybe mock all github calls to make our e2e tests more resilient. WDYT ?

File: marketplace-tests/Cargo.toml [hunk 6]
```diff
@@ -7,6 +7,7 @@ license-file = "LICENSE"
 [lib]
 
 [features]
+default = []
```
Comment (AnthonyBuisset) [inline marketplace-tests/Cargo.toml:10]: why is this needed ?

**Review by AnthonyBuisset — COMMENTED** [review]
coverage tests are failing, I guess because of missing env variables (same as in `ci.yml`)

File: marketplace-tests/Cargo.toml (hunk 2 above)
Comment (tdelabro) [reply marketplace-tests/Cargo.toml:51]: done

File: marketplace-tests/src/e2e_tests/contributors/mod.rs (hunk 1 above)
Comment (tdelabro) [reply marketplace-tests/src/e2e_tests/contributors/mod.rs:4]: done

File: marketplace-tests/src/e2e_tests/scenario/contribution_lifetime.rs (hunk 3 above)
Comment (tdelabro) [reply marketplace-tests/src/e2e_tests/scenario/contribution_lifetime.rs:44]: done

File: marketplace-tests/src/e2e_tests/scenario/contribution_lifetime.rs (hunk 4 above)
Comment (tdelabro) [reply marketplace-tests/src/e2e_tests/scenario/contribution_lifetime.rs:48]: done

File: marketplace-tests/Cargo.toml (hunk 6 above)
Comment (tdelabro) [reply marketplace-tests/Cargo.toml:10]: coz while developing, in order to have the e2e module checked I changed it to `default = ["with_e2e_tests"]`
Now I can just copy past "with_e2e_tests" inside instead of writing the whole thing each time.
